From 6f45e8b863d308946b19219a39bffca062bede0f Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Fri, 29 Sep 2023 19:22:35 +0200
Subject: [PATCH] nrt: Run CRC32 checks

.. and also move fdt and unzip address to avoid
overlaping due to increased kernel size.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 include/configs/imx8mm_var_dart.h | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/include/configs/imx8mm_var_dart.h b/include/configs/imx8mm_var_dart.h
index 07cbc27559..ba725afb2e 100644
--- a/include/configs/imx8mm_var_dart.h
+++ b/include/configs/imx8mm_var_dart.h
@@ -79,10 +79,10 @@
 	"kernel_addr_r=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"bsp_script=boot.scr\0" \
 	"image=Image.gz\0" \
-	"img_addr=0x42000000\0"			\
+	"img_addr=0x436B5300\0"			\
 	"console=ttymxc0,115200\0" \
-	"fdt_addr_r=0x43000000\0" \
-	"fdt_addr=0x43000000\0"			\
+	"fdt_addr_r=0x44B4E200\0" \
+	"fdt_addr=0x44B4E200\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=try\0" \
 	"boot_fit=no\0" \
@@ -122,8 +122,8 @@
 	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=load mmc ${mmcdev}:${resin_root_part} ${img_addr} boot/${image};" \
-		"unzip ${img_addr} ${loadaddr}\0" \
+	"loadimage=if load mmc ${mmcdev}:${resin_root_part} ${img_addr} boot/${image}; then " \
+		"unzip ${img_addr} ${loadaddr}; run balena_kernel_load_crc_save; else false; fi; \0" \
 	"findfdt=" \
 		"if test $fdt_file = undefined; then " \
 			"if test -n ${custom_fdt_file}; then " \
@@ -146,7 +146,7 @@
 		"fi; \0" \
 	"loadfdt=run findfdt; " \
 		"echo fdt_file=${fdt_file}; " \
-		"load mmc ${mmcdev}:${resin_root_part} ${fdt_addr} boot/${fdt_file}\0" \
+		"if load mmc ${mmcdev}:${resin_root_part} ${fdt_addr} boot/${fdt_file}; then run balena_fdt_load_crc_save; else false; fi; \0" \
 	"ramsize_check="\
 		"if test $sdram_size -le 512; then " \
 			"setenv cma_size cma=320M; " \
@@ -164,6 +164,8 @@
 			"bootm ${loadaddr}; " \
 		"else " \
 			"if run loadfdt; then " \
+				"run balena_kernel_load_crc_check; " \
+				"run balena_fdt_load_crc_check; " \
 				"booti ${loadaddr} - ${fdt_addr_r}; " \
 			"else " \
 				"echo WARN: Cannot load the DT; " \
@@ -197,7 +199,7 @@
         "setenv resin_kernel_load_addr ${loadaddr};" \
         "run resin_set_kernel_root; run start_led; run set_os_cmdline; gpio set 137; " \
         "setenv mmcdev ${resin_dev_index};" \
-        "setenv mmcbootpart ${resin_boot_part}; run setconsole;" \
+        "setenv mmcbootpart ${resin_boot_part}; " \
 	"mmc dev ${mmcdev}; "\
 	"if mmc rescan; then " \
 		"if test ${use_m4} = yes && run loadm4bin; then " \
-- 
2.37.2

